gsy_framework/utils.py

```diff
@@ -389,53 +416,68 @@ def deep_size_of(input_obj):
     return memory_size
 
 
-def utf8len(s):
-    return len(s.encode('utf-8'))
+def utf8len(string: str) -> int:
+    """Return the length of a string encoded in UTF-8 format."""
+    return len(string.encode("utf-8"))
 
 
-def get_json_dict_memory_allocation_size(json_dict):
+def get_json_dict_memory_allocation_size(json_dict: Dict):
+    """Return the memory allocation of the provided dictionary."""
     return utf8len(json.dumps(json_dict)) / 1024.
 
 
-def area_name_from_area_or_iaa_name(name):
-    return name[4:] if name[:4] == 'IAA ' else name
+def area_name_from_area_or_ma_name(name: str) -> str:
+    """Retrieve the name of the area.
+
+    The input can be either the area name itself or the name of the market agent of the area.
+    """
+    return name[3:] if name[:3] == "MA " else name
```

Comment:
Finally: since `MA` is shorter than `IAA`, this check now uses 3 characters rather than 4.